components/cop_datatype/src/def/eval_type.rs

```diff
@@ -66,6 +66,8 @@ impl std::convert::TryFrom<crate::FieldTypeTp> for EvalType {
             | crate::FieldTypeTp::LongBlob
             | crate::FieldTypeTp::Blob
             | crate::FieldTypeTp::VarString
+            | crate::FieldTypeTp::Enum
+            | crate::FieldTypeTp::Set
```

Comment:
You should not do it here because it affect not only table / index scans, but also all executors.